addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -253,62 +255,156 @@ def test_post_wo_access(self):
+            record.activity_schedule("test_mail.mail_act_test_todo", user_id=user_employee.id)
+            for record in test_records_all
+        ))
+        employee_acts, employee_acts_on_admin_rec = employee_acts_all[:2], employee_acts_all[2:]
+
+        self.assertTrue((company_1_all | company_2_admin_only) <= user_admin.company_ids)
+        self.assertEqual(company_1_all, user_employee.company_ids)
+
+        # We test the outcome of systray_get_activities for different couple of user and allowed companies
+        for description, (user, allowed_company_ids), (expected_other_activities, expected_test_model_activities) in (
+                (
+                        "Admin see only activities of records of allowed companies (company_1).",
+                        (user_admin, company_1_all.ids),
+                        (False, admin_acts[0] + admin_acts_on_admin_rec[0] + admin_acts_on_admin_rec[2]),
+                ),
+                (
+                        "Admin see only activities of records of allowed companies (company_2).",
+                        (user_admin, company_2_admin_only.ids),
+                        (False, admin_acts[1] + admin_acts_on_admin_rec[1] + admin_acts_on_admin_rec[2]),
+                ),
+                (
+                        "Admin see only activities of records of allowed companies (company_1 and company_2).",
+                        (user_admin, (company_1_all | company_2_admin_only).ids),
+                        (False, admin_acts_all),
+                ),
+                (
+                        'Employee see all activities of records of allowed companies (company_1) he has access to, '
+                        'and under "Other activities", see all activities of allowed companies he has not access to'
+                        ' + activities related to record with company False or he has not access to.',
+                        (user_employee, company_1_all.ids),
+                        # Employee has no right on test_records_admin so related activities appears in "other"
+                        # He has no access to company_2 so activity of test_records[1] is also included
+                        (employee_acts[1] + employee_acts_on_admin_rec, employee_acts[0]),
+                ),
+        ):
+            with self.subTest(description=description):
+                self.authenticate(user.login, user.login)
+                with patch.object(MailTestMultiCompanyWithActivity, '_filter_access_rules', autospec=True,
+                                  side_effect=_mock_filter_access_rules):
+                    activity_groups = self.make_jsonrpc_request("/mail/data", {
+                        "systray_get_activities": True,
+                        "context": {"allowed_company_ids": allowed_company_ids}
+                    })["Store"]["activityGroups"]
+                actual_activity_groups_by_model = {ag["model"]: ag for ag in activity_groups if ag["model"]}
```

Comment:
Would remove "actual" to simplify.

As the others are prefixed "expected", it's sufficient.